examples/air-quality-sensor-app/linux/AirQualitySensorAppAttrUpdateDelegate.cpp

```diff
@@ -0,0 +1,313 @@
+
+    if (clusterName == "AirQuality")
+    {
+        uint8_t newValue = static_cast<uint8_t>(self->mJsonValue["NewValue"].asUInt());
+        self->OnAirQualityChangeHandler(newValue);
+    }
+    else if (clusterName == "TemperatureMeasurement")
+    {
+        int16_t newValue = static_cast<int16_t>(self->mJsonValue["NewValue"].asUInt());
+        self->OnTemperatureChangeHandler(newValue);
+    }
+    else if (clusterName == "RelativeHumidityMeasurement")
+    {
+        uint16_t newValue = static_cast<uint16_t>(self->mJsonValue["NewValue"].asUInt());
+        self->OnHumidityChangeHandler(newValue);
+    }
+    else if (clusterName.find("Concentration") != std::string::npos)
+    {
+        float newValue = static_cast<uint16_t>(self->mJsonValue["NewValue"].asFloat());
+        self->OnConcetratorChangeHandler(clusterName, newValue);
+    }
+    else
+    {
+        ChipLogError(NotSpecified, "Invalid cluster name %s", clusterName.c_str());
+    }
+
+    // Delete AirQualitySensorAttrUpdateHandler
+    Platform::Delete(self);
+}
+
+void AirQualitySensorAttrUpdateHandler::OnAirQualityChangeHandler(uint8_t newValue)
+{
+    EndpointId endpoint  = 1;
+    EmberAfStatus status = AirQuality::Attributes::AirQuality::Set(endpoint, static_cast<AirQuality::AirQualityEnum>(newValue));
+    VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(NotSpecified, "Failed to set AirQuality attribute"));
+    ChipLogDetail(NotSpecified, "The new AirQuality value: %d", newValue);
+}
+
+void AirQualitySensorAttrUpdateHandler::OnTemperatureChangeHandler(int16_t newValue)
+{
+    EndpointId endpoint  = 1;
+    EmberAfStatus status = TemperatureMeasurement::Attributes::MeasuredValue::Set(endpoint, newValue);
+    VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status,
+                   ChipLogError(NotSpecified, "Failed to TemperatureMeasurement MeasuredValue attribute"));
+    ChipLogDetail(NotSpecified, "The new TemperatureMeasurement value: %d", newValue);
+
+    DataModel::Nullable<int16_t> minVal;
+    TemperatureMeasurement::Attributes::MinMeasuredValue::Get(endpoint, minVal);
+    if (minVal.IsNull())
+    {
+        status = TemperatureMeasurement::Attributes::MinMeasuredValue::Set(endpoint, newValue);
+    }
+    else if (newValue < minVal.Value())
+    {
+        status = TemperatureMeasurement::Attributes::MinMeasuredValue::Set(endpoint, newValue);
+        VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status,
+                       ChipLogError(NotSpecified, "Failed to set TemperatureMeasurement MinMeasuredValue attribute"));
+    }
+
+    DataModel::Nullable<int16_t> maxVal;
+    TemperatureMeasurement::Attributes::MaxMeasuredValue::Get(endpoint, maxVal);
+    if (maxVal.IsNull())
+    {
+        status = TemperatureMeasurement::Attributes::MaxMeasuredValue::Set(endpoint, newValue);
+    }
+    else if (newValue > maxVal.Value())
+    {
+        status = TemperatureMeasurement::Attributes::MaxMeasuredValue::Set(endpoint, newValue);
```

Comment:
I meant to say, I think you are using MaxMeasuredValue wrong. You seem to be updating it each time a new sensor reading comes in but it's supposed to represent the absolute min and max that the sensor can read. 

> This attribute SHALL represent the maximum value of MeasuredValue that is capable of being
measured. A MaxMeasuredValue of null indicates that the MaxMeasuredValue is not defined.